src/Sharpbrake.NLog/AirbrakeTarget.cs

```diff
@@ -17,67 +18,84 @@ public class AirbrakeTarget : TargetWithLayout
         /// This key is what you configure the notifier agent in your app to use.
         /// </summary>
         [RequiredParameter]
-        public string ProjectId { get; set; }
+        public string ProjectId { get => (_projectId as SimpleLayout)?.Text; set => _projectId = value ?? string.Empty; }
+        private Layout _projectId;
 
         /// <summary>
         /// Host name of the endpoint.
         /// </summary>
-        public string Host { get; set; }
+        public string Host { get => (_host as SimpleLayout)?.Text; set => _host = value ?? string.Empty; }
+        private Layout _host;
 
         /// <summary>
         /// Path to logging file.
         /// </summary>
-        public string LogFile { get; set; }
+        public string LogFile { get => (_logFile as SimpleLayout)?.Text; set => _logFile = value ?? string.Empty; }
+        private Layout _logFile;
 
         /// <summary>
         /// Represents URI string of proxy provider.
         /// </summary>
-        public string ProxyUri { get; set; }
+        public string ProxyUri { get => (_proxyUri as SimpleLayout)?.Text; set => _proxyUri = value ?? string.Empty; }
+        private Layout _proxyUri;
 
         /// <summary>
         /// The credentials for proxy.
         /// </summary>
-        public string ProxyUsername { get; set; }
+        public string ProxyUsername { get => (_proxyUsername as SimpleLayout)?.Text; set => _proxyUsername = value ?? string.Empty; }
+        private Layout _proxyUsername;
 
         /// <summary>
         /// The credentials for proxy.
         /// </summary>
-        public string ProxyPassword { get; set; }
+        public string ProxyPassword { get => (_proxyPassword as SimpleLayout)?.Text; set => _proxyPassword = value ?? string.Empty; }
+        private Layout _proxyPassword;
 
         /// <summary>
         /// Comma-separated list of environments that should be ignored.
         /// </summary>
-        public string IgnoreEnvironments { get; set; }
+        public string IgnoreEnvironments { get => (_ignoreEnvironments as SimpleLayout)?.Text; set => _ignoreEnvironments = value ?? string.Empty; }
+        private Layout _ignoreEnvironments;
 
         /// <summary>
         /// Comma-separated list of parameters which will not be filtered.
         /// If count of parameters is not zero - all not listed parameters will be filtered.
         /// </summary>
-        public string WhitelistKeys { get; set; }
+        public string WhitelistKeys { get => (_whitelistKeys as SimpleLayout)?.Text; set => _whitelistKeys = value ?? string.Empty; }
+        private Layout _whitelistKeys;
 
         /// <summary>
         /// Comma-separated list of parameters which will be filtered out.
         /// </summary>
-        public string BlacklistKeys { get; set; }
+        public string BlacklistKeys { get => (_blacklistKeys as SimpleLayout)?.Text; set => _blacklistKeys = value ?? string.Empty; }

```

Comment:
looks good, but maybe it would be nice to have helpers for the get/setters (to avoid boilerplate)

Reply:
The correct solution would be to use Layout, but that would be a breaking change.